orbit/pkg/bitlocker/bitlocker_management_windows.go

```diff
@@ -0,0 +1,574 @@
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return fmt.Errorf("decrypt(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	return nil
+}
+
+// prepareVolume prepares a new Bitlocker Volume. This should be called BEFORE any key protectors are added.
+// Example: vol.prepareVolume(bitlocker.VolumeTypeDefault, bitlocker.EncryptionTypeHardware)
+// https://docs.microsoft.com/en-us/windows/win32/secprov/preparevolume-win32-encryptablevolume
+func (v *Volume) prepareVolume(volType DiscoveryVolumeType, encType ForceEncryptionType) error {
+	resultRaw, err := oleutil.CallMethod(v.handle, "PrepareVolume", string(volType), int32(encType))
+	if err != nil {
+		return fmt.Errorf("prepareVolume(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return fmt.Errorf("prepareVolume(%s): %w", v.letter, encryptErrHandler(val))
+	}
+	return nil
+}
+
+// protectWithNumericalPassword adds a numerical password key protector.
+// Leave password as a blank string to have one auto-generated by Windows
+// https://docs.microsoft.com/en-us/windows/win32/secprov/protectkeywithnumericalpassword-win32-encryptablevolume
+func (v *Volume) protectWithNumericalPassword() (string, error) {
+	var volumeKeyProtectorID ole.VARIANT
+	ole.VariantInit(&volumeKeyProtectorID)
+	var resultRaw *ole.VARIANT
+	var err error
+
+	resultRaw, err = oleutil.CallMethod(v.handle, "ProtectKeyWithNumericalPassword", nil, nil, &volumeKeyProtectorID)
+	if err != nil {
+		return "", fmt.Errorf("ProtectKeyWithNumericalPassword(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return "", fmt.Errorf("ProtectKeyWithNumericalPassword(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	var recoveryKey ole.VARIANT
+	ole.VariantInit(&recoveryKey)
+	resultRaw, err = oleutil.CallMethod(v.handle, "GetKeyProtectorNumericalPassword", volumeKeyProtectorID.ToString(), &recoveryKey)
+
+	if err != nil {
+		return "", fmt.Errorf("GetKeyProtectorNumericalPassword(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return "", fmt.Errorf("GetKeyProtectorNumericalPassword(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	return recoveryKey.ToString(), nil
+}
+
+// protectWithPassphrase adds a passphrase key protector
+// https://docs.microsoft.com/en-us/windows/win32/secprov/protectkeywithpassphrase-win32-encryptablevolume
+func (v *Volume) protectWithPassphrase(passphrase string) error {
+	var volumeKeyProtectorID ole.VARIANT
+	ole.VariantInit(&volumeKeyProtectorID)
+
+	resultRaw, err := oleutil.CallMethod(v.handle, "ProtectKeyWithPassphrase", nil, passphrase, &volumeKeyProtectorID)
+	if err != nil {
+		return fmt.Errorf("protectWithPassphrase(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return fmt.Errorf("protectWithPassphrase(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	volumeKeyProtectorID.ToString()
```

Comment:
This seems like something that needed to be removed or stored somewhere?

Reply:
Thanks, I've just fixed this. This function is not currently used. I've left the logic implemented for future usecase (fixed drives encryption)